internal/tofu/node_resource_abstract_instance.go

```diff
@@ -346,6 +346,54 @@ func (n *NodeAbstractResourceInstance) writeResourceInstanceStateImpl(ctx EvalCo
 	return nil
 }
 
+// planForget returns a removed from state diff.
+func (n *NodeAbstractResourceInstance) planForget(ctx EvalContext, currentState *states.ResourceInstanceObject, deposedKey states.DeposedKey) (*plans.ResourceInstanceChange, tfdiags.Diagnostics) {
+	var diags tfdiags.Diagnostics
+	var plan *plans.ResourceInstanceChange
+
+	// TODO: remove - this is an unreachable code, the state of this resource cannot be null
```

Comment:
I'll remove it before I merge the PR. This is an unreachable code because this scenario doesn't make sense. The forgotten resources must be orphans or deposed, which means they exist in the state.
But I'm still trying to determine if I missed any relevant scenario.

Reply:
Removed it.